### PR TITLE
Plug mempool to libp2p backend

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

File: nomos-services/mempool/src/network/adapters/libp2p.rs [hunk 1]
```diff
@@ -24,15 +30,39 @@ where
...
+                sender,
+            })
+            .await
+            .expect("Network backend should be ready");
+        let receiver = receiver.await.unwrap();
+        Box::new(Box::pin(BroadcastStream::new(receiver).filter_map(
+            move |message| match message {
+                Ok(Event::Message(Message { data, topic, .. })) if topic == topic_hash => {
+                    let tx = wire::deserialize::<TransactionMsg<Tx>>(&data)
+                        .expect("Failed to deserialize tx");
```
Comment (Zeegomo) [inline nomos-services/mempool/src/network/adapters/libp2p.rs:61]: we probably don't want to panic on this

**Review by Zeegomo — APPROVED** [review]
Good after comment is fixed

File: nomos-services/mempool/src/network/adapters/libp2p.rs (hunk 1 above)
Comment (danielSanchezQ) [reply nomos-services/mempool/src/network/adapters/libp2p.rs:61]: Oh yes, I left for placeholder as I was having some other issue, and forgot to remove. Thanks!!

**Review by youngjoon-lee — COMMENTED** [review]
lgtm

**Pull request merged**